### ISSUE TITLE
br lint ignores the acceptance_criteria field and passes on any description containing the words

### ISSUE THREAD

**Issue by joshuajbrunner** [Dicklesworthstone/beads_rust#509]
## Summary

`br lint` checks for required sections by substring-searching the `description` column. It never reads `acceptance_criteria`, the column every documented command writes to.

This fails in both directions:

- an issue with criteria in the field is reported as **missing** them
- an issue with **no** criteria passes, provided its description mentions the words anywhere

Confirmed on `main` at `31948e5` and in the released `br 0.6.0`.

## The case that shows it best

A control issue, created deliberately without criteria to check that lint would catch it:

```
Control issue intentionally created without acceptance criteria to verify
that `br lint` detects the missing field.

## Steps to Reproduce
1. View this issue and confirm its acceptance-criteria field is absent.
...
```

Type `bug`, so lint requires both `## Steps to Reproduce` and `## Acceptance Criteria`. Its `acceptance_criteria` column is empty.

`br lint` passes it, because the sentence *explaining that it has no acceptance criteria* contains the words "acceptance criteria".

## Minimal reproduction

```sh
br init --prefix t

# criteria in the documented field
br create "criteria in the field" -t task \
  --acceptance-criteria "- [ ] the thing works"

# no criteria, but the phrase appears in prose
br create "no criteria at all" -t task \
  -d 'We will add acceptance criteria later.'

br lint
```

**Actual:**

```
Template warnings (1 issues, 1 warnings):

t-4h8 [task]: criteria in the field
  ⚠ Missing: ## Acceptance Criteria - Define criteria to verify completion
```

Exit `1`. The issue with criteria is flagged; the issue without passes.

Database state confirms both were stored as intended:

```
t-4h8 | (empty description)                    | - [ ] the thing works
t-vhr | We will add acceptance criteria later.  | (empty)
```

**Expected:** the reverse.

## Cause

`src/cli/commands/lint.rs:425`

```rust
let description = issue.description.as_deref().unwrap_or("");
let missing = missing_sections(description, required);
```

`missing_sections` (line 468) lowercases the description and calls `.contains()`. `strip_heading_prefix` removes `## ` before comparing, so it is not even a heading match: any occurrence of the words satisfies it.

`acceptance_criteria` appears in that file only in tests.

## The projection looks deliberate

`lint_issues_with_storage_matches_full_hydration_results` (line 580) sets up an issue as:

```rust
missing.acceptance_criteria = Some("unused criteria".repeat(512));
...
assert!(projected_issue.acceptance_criteria.is_none());
```

The fixture names the value "unused criteria" and asserts lint projects it away. The narrow projection presumably exists to avoid hydrating large text columns, but it discards the one column the check is about.

## Why this is worth fixing rather than working around

Every other part of `br` treats `acceptance_criteria` as the home for this data: the schema column, `--acceptance-criteria` on create and update, `--add-acceptance`, `--check-acceptance` with per-item validation and byte-for-byte preservation, `--force` guarding destructive rewrites, `br show`'s rendering, the JSONL export, and `require_acceptance_criteria_satisfied` in #503.

`br lint` is the sole consumer looking elsewhere. Since it exits `1`, it is a natural pre-commit gate, but gating on it today rewards writing the phrase into prose and penalises using the documented flags.

## Suggested fix

Include `acceptance_criteria` in the projection and treat the section as present when the field is non-empty, falling back to the description scan for issues written before the column existed.

Separately, the description scan could require a line-initial heading rather than any substring, which would fix the false pass on its own.

## Possibly related, in beads_viewer

In two separate workspaces, criteria written through `br update --add-acceptance` and `br create --acceptance-criteria` are stored and exported correctly but are not displayed in `bv`, while an issue carrying the same text inside its `description` does display. The viewer's source appears to read the field, so the cause is not yet identified and this is not part of this report. Noted only because both tools diverge from the same column, and a reader of this issue may hit the viewer symptom while verifying a fix. Happy to file separately in `beads_viewer` once the cause is pinned down.

## Related

#503, same column, different consumer, and evidence that the field is the intended source of truth.

**Comment by Dicklesworthstone** [comment]
Thanks for the clean reproduction; the control-issue case says it all. This was fixed on `main` the day you filed it, in 3e5ca281 and 404b7813 (2026-09-19), which postdate the v0.6.0 tag, so it is not in a release yet.

- `br lint` now reads the `acceptance_criteria` field: a non-blank field satisfies "Acceptance Criteria" for tasks and features on its own, even with an empty description, and the bounded workspace scan hydrates the field for every linted issue instead of dropping it in the projection.
- Legacy section detection no longer accepts prose: it requires a real Markdown heading (`## Acceptance Criteria`, case-insensitive) with a non-empty body, and headings inside fenced code blocks do not count. "We will add acceptance criteria later." no longer passes.
- The field does not replace `## Steps to Reproduce` for bugs or success criteria for epics, so the bug-type control issue in your report is still flagged for its missing steps.

I re-ran the lint suite at main today (18 tests, including `acceptance_field_satisfies_tasks_and_features_without_description` and `legacy_sections_reject_prose_lookalikes_and_empty_bodies`, which are your two cases). Closing; it will be in the next tagged release.
